crates/server/src/authority/auth_lookup.rs

```diff
@@ -363,17 +355,37 @@ impl<'a> IntoIterator for &'a LookupRecords {
         match self {
             LookupRecords::Empty => LookupRecordsIter::Empty,
             LookupRecords::Records {
-                is_secure,
-                supported_algorithms,
+                lookup_options,
                 records,
-            } => LookupRecordsIter::RecordsIter(records.records(*is_secure, *supported_algorithms)),
-            LookupRecords::ManyRecords(is_secure, supported_algorithms, r) => {
```

Comment:
There's quite a bit of repetition here. Can we extract the `LookupRecordsIter::ManyRecordsIter` out of the `cfg_if!()` call, and maybe also add a convenience method (either as method to `LookupOptions` or a local helper function here) that can support the different scenarios here?

Reply:
Thanks for that feedback. Let me see about deduping this.

Reply:
Ok, great suggestion. I think this is much cleaner now.